$NetBSD: patch-ae,v 1.10 2023/11/23 16:15:04 ryoon Exp $

 - Don't assert GNU ld on Solaris, where it might not be true.
 - Add endian compat.

--- include/local.h.orig	2023-05-04 07:16:16.000000000 +0000
+++ include/local.h
@@ -45,6 +45,15 @@
 #endif
 #ifndef __BIG_ENDIAN
 #define __BIG_ENDIAN BIG_ENDIAN
+#elif defined(__sun)
+#include <sys/byteorder.h>
+#define __LITTLE_ENDIAN 1234
+#define __BIG_ENDIAN 4321
+#ifdef _BIG_ENDIAN
+#define __BYTE_ORDER __BIG_ENDIAN
+#else
+#define __BYTE_ORDER __LITTLE_ENDIAN
+#endif
 #endif
 #include <stdarg.h>
 #include <poll.h>
@@ -274,7 +283,9 @@ extern snd_lib_error_handler_t snd_err_m
 
 /*
  */
+#ifndef __sun
 #define HAVE_GNU_LD
+#endif
 #define HAVE_ELF
 #define HAVE_ASM_PREVIOUS_DIRECTIVE
 
